src/NUnitEngine/nunit.engine.core/Internal/AddinsFileReader.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using NUnit.Engine.Internal.FileSystemAccess;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace NUnit.Engine.Internal
+{
+    /// <summary>
+     /// A reader for NUnit addins-files.
+     /// </summary>
+     /// <remarks>
+     /// The format of an addins-file can be found at https://docs.nunit.org/articles/nunit-engine/extensions/Installing-Extensions.html.
+     /// </remarks>
+    internal sealed class AddinsFileReader : IAddinsFileReader
+    {
+        /// <inheritdoc/>
+        public IEnumerable<string> Read(IFile file)
+        {
+            if (file == null)
+            {
+                throw new ArgumentNullException(nameof(file));
+            }
+
+            using (var reader = new FileStream(file.FullName, FileMode.Open,  FileAccess.Read, FileShare.Read))
+            {
+                return this.Read(reader);
+            }
+        }
+
+        /// <summary>
+        /// Reads the content of an addins-file from a stream.
+        /// </summary>
+        /// <param name="stream">Input stream. Must be readable and positioned at the beginning of the file.</param>
+        /// <returns>All entries contained in the file.</returns>
+        /// <exception cref="System.IO.IOException"><paramref name="stream"/> cannot be read</exception>
+        internal IEnumerable<string> Read(Stream stream)
+        {
+            var result = new List<string>();
+            using (var reader = new StreamReader(stream))
+            {
+                for(var line = reader.ReadLine(); line != null; line = reader.ReadLine())
+                {
+                    line = line.Split(new char[] { '#' })[0].Trim();
+                    if (line != string.Empty)
+                    {
+                        result.Add(line.Replace('\\', '/'));
```

Comment:
Observation: this is a change in logic I think - previously the substitution would only happen on windows systems. I can't think of an issue that would cause - unless someone has a valid unix path with a backslash in...which would be very weird!

Thinking about it, is the substitution even necessary? I would have thought the .NET path classes would have taken care of that.

Reply:
My first thought was that the substitution is harmless on non-Windows systems and that's why I changed the behavior.
But when I read your comment I remembered that macOS uses backslashes to escape special character in paths (e.g. ' '). I will change the behavior back to the original. 🤦‍♂️

Reply:
I reverted the original behavior.

@ChrisMaddock: You are right there is no difference in behavior on Windows-systems between the [DirectorySeparatorChar](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.directoryseparatorchar?view=net-5.0) and the [AltDirectorySeparatorChar](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.altdirectoryseparatorchar?view=net-5.0). The only advantage I see by replacing the backslash is that you would only have to consider one character during development.